File: sql/mysql/Go/SqlModes.go
package parser

import (
    "strings"
)

func sqlModeFromString(modes string) map[SqlMode]bool {
	result := make(map[SqlMode]bool)
	parts := strings.Split(strings.ToUpper(modes), ",")

	for _, mode := range parts {
		switch mode {
		case "ANSI", "DB2", "MAXDB", "MSSQL", "ORACLE", "POSTGRESQL":
			result[AnsiQuotes] = true
			result[PipesAsConcat] = true
			result[IgnoreSpace] = true
		case "ANSI_QUOTES":
			result[AnsiQuotes] = true
		case "PIPES_AS_CONCAT":
			result[PipesAsConcat] = true
		case "NO_BACKSLASH_ESCAPES":
			result[NoBackslashEscapes] = true
		case "IGNORE_SPACE":
			result[IgnoreSpace] = true
		case "HIGH_NOT_PRECEDENCE", "MYSQL323", "MYSQL40":
			result[HighNotPrecedence] = true
		}
	}

	return result
}

File: sql/mysql/JavaScript/SqlModes.js
/** SQL modes that control parsing behavior. */
import SqlMode from "./SqlMode.js";
export class SqlModes {
    /**
     * Converts a mode string into individual mode flags.
     *
     * @param modes The input string to parse.
     */
    static sqlModeFromString(modes) {
        var result = new Set();
        const parts = modes.toUpperCase().split(",");
        parts.forEach((mode) => {
            if (mode === "ANSI" || mode === "DB2" || mode === "MAXDB" || mode === "MSSQL" || mode === "ORACLE" ||
                mode === "POSTGRESQL") {
                result.add(SqlMode.AnsiQuotes).add(SqlMode.PipesAsConcat).add(SqlMode.IgnoreSpace);
            }
            else if (mode === "ANSI_QUOTES") {
                result.add(SqlMode.AnsiQuotes);
            }
            else if (mode === "PIPES_AS_CONCAT") {
                result.add(SqlMode.PipesAsConcat);
            }
            else if (mode === "NO_BACKSLASH_ESCAPES") {
                result.add(SqlMode.NoBackslashEscapes);
            }
            else if (mode === "IGNORE_SPACE") {
                result.add(SqlMode.IgnoreSpace);
            }
            else if (mode === "HIGH_NOT_PRECEDENCE" || mode === "MYSQL323" || mode === "MYSQL40") {
                result.add(SqlMode.HighNotPrecedence);
            }
        });
        return result;
    }
}
export default SqlModes;
